Source/Dafny/Verifier/Translator.cs

```diff
@@ -11142,6 +11142,12 @@ public enum K { Free, Checked, Both }
         }
 
         return true;
+
+      } else if (expr is MatchExpr) {
+        var e = (MatchExpr)expr;
+        var ite = etran.DesugarMatchExpr(e);
```

Comment:
Shouldn't `DesugareMatchExpr` be done in an earlier phase?

Reply:
Thanks for the suggestion. I looked into it, but my conclusion is "no". Although TrExpr, TrSplitExpr, and CanCallAssumptions all do the desugaring, CheckWellformedness starts from the sugared version (because it wants to check the well-formedness of the match-source expression once, not just per `case`).